zppy/templates/tc_analysis.bash

```diff
@@ -0,0 +1,118 @@
+#!/bin/bash
+{% include 'slurm_header.sh' %}
+
+{{ environment_commands }}
+
+set -e # Stop running script on error
+
+# Turn on debug output if needed
+debug={{ debug }}
+if [[ "${debug,,}" == "true" ]]; then
+  set -x
+fi
+
+# Script dir
+cd {{ scriptDir }}
+
+# Get jobid
+id=${SLURM_JOBID}
+
+# Update status file
+STARTTIME=$(date +%s)
+echo "RUNNING ${id}" > {{ prefix }}.status
+
+# A Bash script to post-process E3SM 6 hourly (h2) instantaneous output to generate a text file storing Tropical Cyclone tracks
+# tempestremap and tempestextremes are built in e3sm-unified from version 1.5.0.
+
+#For typical EAM v2 ne gp2 grids.
+start="{{ '%04d' % (year1) }}"
+end="{{ '%04d' % (year2) }}"
+caseid="{{ case }}"
+drc_in={{ input }}/{{ input_subdir }}
+# Warning: tempest-remap can only write grid file on SCRATCH space.
+# The result files will be moved to another path at the end.
+result_dir_fin={{ output }}/post/atm # Directory will be {{ output }}/post/atm/tc-analysis_${Y1}_${Y2}
+mkdir -p $result_dir_fin
+y1={{ year1 }}
+y2={{ year2 }}
+Y1="{{ '%04d' % (year1) }}"
+Y2="{{ '%04d' % (year2) }}"
+result_dir={{ scratch }}/tc-analysis_${Y1}_${Y2}/
+
+atm_name={{ atm_name }}
+
+# Determine res and pg2
+first_file=`echo $(ls ${drc_in}/${caseid}.{{ input_files }}.*.nc | head -n 1)`
+topography_file=`echo $(ncks --trd -M -m ${first_file} | grep -E -i "^global attribute [0-9]+: topography_file" | cut -f 11- -d ' ' | sort)`
+res=""
+pg2=false
+if [[ $topography_file =~ /.*_(.*)_.*nc ]]; then
+    grid=${BASH_REMATCH[1]}
+    if [[ $grid =~ ne([0-9]*) ]]; then
+	res=${BASH_REMATCH[1]}
+    fi
+    if [[ $grid =~ pg2 ]]; then
+	pg2=true
+    fi
+fi
+
+mkdir -p $result_dir
+file_name=${caseid}_${start}_${end}
+
+# Generate mesh files (.g).
+GenerateCSMesh --res $res --alt --file ${result_dir}outCSne$res.g
+out_type="CGLL"
+# For v2 production simulation with pg2 grids:
+if $pg2; then
+    GenerateVolumetricMesh --in ${result_dir}outCSne$res.g  --out ${result_dir}outCSne$res.g --np 2 --uniform
+    out_type="FV"
+fi
+echo $out_type
+# Generate connectivity files (.dat)
+GenerateConnectivityFile --in_mesh ${result_dir}outCSne$res.g --out_type $out_type --out_connect ${result_dir}connect_CSne${res}_v2.dat
```

Comment:
@chengzhuzhang This command fails with `GenerateConnectivityFile: error while loading shared libraries: libnetcdf.so.11: cannot open shared object file: No such file or directory` if two year sets are run simultaneously (e.g., `years = "1985-1990", "1985-1994",`). If I run the bash files for each year set separately, I don't get this error. I thought changing `result_dir` from `tc-analysis`  to `tc-analysis_${Y1}_${Y2}` would fix this, but apparently not.

Do you have an idea of what's going on here?

Reply:
Hey Ryan, is this an issue that, two concurrent `GenerateConnectivityFile` calls from `e3sm_unified` would cause this error? Even if they are operate on files in separate folders? I'm pinging @paullric to take a look at this error message..

Reply:
I suppose that's possible, but I don't see why two concurrent calls would cause an issue.